### ISSUE TITLE
clang-format with `-lines` removes space before `{`

### ISSUE THREAD

**Issue by zmodem** [llvm/llvm-project#67660]
Consider:

```
$ cat /tmp/a.cc
TaskRunnerProvider::TaskRunnerProvider(
    scoped_refptr<base::SingleThreadTaskRunner> main_task_runner,
    scoped_refptr<base::SingleThreadTaskRunner> impl_task_runner)
#if FOO
    : main_task_runner_(main_task_runner), impl_task_runner_(impl_task_runner) {
#else
    : main_task_runner_(main_task_runner),
      foo_(false) {
#endif
}

TaskRunnerProvider::~TaskRunnerProvider() {
}
```

Running clang-format on only the `foo_(false) {` line removes the space before `{`:

```
$ build/bin/clang-format -lines=8:8 /tmp/a.cc
TaskRunnerProvider::TaskRunnerProvider(
    scoped_refptr<base::SingleThreadTaskRunner> main_task_runner,
    scoped_refptr<base::SingleThreadTaskRunner> impl_task_runner)
#if FOO
    : main_task_runner_(main_task_runner), impl_task_runner_(impl_task_runner) {
#else
    : main_task_runner_(main_task_runner),
      foo_(false){
#endif
}

TaskRunnerProvider::~TaskRunnerProvider() {
}
```

**Comment by rsniezek** [comment]
I have a bit differentissue with the --lines switch. 
For me it doesn't remove extra spaces of indentation. e.g. I have an IF block, where one (or both) lines have extra spaces after tab, 
```cpp
	if (condition) {
		    do_something();
		return 0;
	}
```
When I run clang-format on entire file, the block correctly gets formatted as
```cpp
	if (condition) {
		do_something();
		return 0;
	}
```
but when I restrict the formatting with --lines, making sure that the culprit line is in the scope, the spaces are NOT removed, or sometimes they are even added to the second line and I end up with
```cpp
	if (condition) {
		    do_something();
		    return 0;
	}
```
Not sure what's the triggering scenario but have identical behavior with git-clang-format, which makes it pretty much unusable in legacy codebases. clang-format in version 16.0.6

**Comment by owenca** [comment]
Fixed by 063e3fe64853. @rsniezek can you file a separate issue with yours?

**Comment by rsniezek** [comment]
@owenca will do once I extract minimal reproduction example.
